DRAFT.md

```diff
@@ -848,186 +864,191 @@ private:
+  If `other` is valueless, `*this` is valueless. Otherwise, constructs an owned object of
+  type `T` with `*other`, using the allocator `alloc`.
 
 ```c++
 constexpr indirect(allocator_arg_t, const Allocator& a,
                    const indirect& other);
 ```
 
-11. _Mandates_: `T` is a complete type.
+9. _Mandates_: `T` is a complete type.
 
-12. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
+10. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
     valueless, `*this` is valueless. Otherwise, constructs an owned object of
     type `T` with `*other`, using the allocator `alloc`.
 
 ```c++
 constexpr indirect(indirect&& other) noexcept;
 ```
 
-13. _Mandates_: If `allocator_traits<allocator_type>::is_always_equal::value`
+11. _Mandates_: If `allocator_traits<Allocator>::is_always_equal::value`
     is `false` then `T` is a complete type.
 
-14. _Effects_: If `other` is valueless, `*this` is valueless. Otherwise, if
-    `alloc == other.alloc` is `true` constructs an object of type `indirect`
-    that owns the owned object of `other`; `other` is valueless. Otherwise,
-    constructs an owned object of type `T` with `*std::move(other)`, using the
-    allocator `alloc`.
+12. _Effects_: `alloc` is direct-non-list initialised from `std::move(other.alloc)`.
+  If `other` is valueless, `*this` is valueless. Otherwise `*this` takes
+  ownership of the owned object of `other`.
+
+13. _Postconditions_: `other` is valueless.
 
 ```c++
 constexpr indirect(allocator_arg_t, const Allocator& a, indirect&& other)
    noexcept(allocator_traits<Allocator>::is_always_equal::value);
 ```
 
-15. _Mandates_: If `allocator_traits<allocator_type>::is_always_equal::value`
+14. _Mandates_: If `allocator_traits<Allocator>::is_always_equal::value`
     is `false` then `T` is a complete type.
 
-16. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
-    valueless, `*this` is valueless. Otherwise, if `alloc == other.alloc` is `true`
+15. _Effects_: `alloc` is direct-non-list-initialized with `a`. If `other` is
+    valueless, `*this` is valueless. Otherwise, if `alloc == other.alloc` is `true`,
     constructs an object of type `indirect` that owns the owned object of `other`;
     `other` is valueless. Otherwise, constructs an owned object of type `T` with
     `*std::move(other)`, using the allocator `alloc`.
 
+16. _Postconditions_: `other` is valueless.
+
 ```c++
-template <class... Us>
-explicit constexpr indirect(in_place_t, Us&&... us);
+template <class U=T>
+explicit constexpr indirect(U&& u);
 ```
 
-17. _Constraints_: `is_constructible_v<T, Us...>` is `true`.
-   `is_copy_constructible_v<T>` is `true`.
-   `is_default_constructible_v<allocator_type>` is `true`.
+17. _Constraints_: Where `UU` is `remove_cvref_t<U>`,
+    * `is_same_v<UU, indirect>` is `false`,
+    * `is_same_v<UU, in_place_t>` is `false`,
+    * `is_constructible_v<T, UU>` is `true`,
```

Comment:
This should be `is_constructible_v<T, U>`

Reply:
Is updated in PR.